### ISSUE TITLE
Chapter 5 - Text Generation | Beam Search Decoding - Log Probabilities

### ISSUE THREAD

**Issue by gcmsrc** [nlp-with-transformers/notebooks#67]
## Information

The question or comment is about chapter:

* [ ] Introduction
* [ ] Text Classification
* [ ] Transformer Anatomy
* [ ] Multilingual Named Entity Recognition
* [x] Text Generation
* [ ] Summarization
* [ ] Question Answering
* [ ] Making Transformers Efficient in Production
* [ ] Dealing with Few to No Labels
* [ ] Training Transformers from Scratch
* [ ] Future Directions

## Question or comment
In section **Beam Search Decoding** of chapter 5, at page 132, the Authors include the following function for calculating a sequence log-probability:

```
def sequence_logprob(model, labels, input_len=0):
    with torch.no_grad():
        output = model(labels)
        log_probs = log_probs_from_logits(
            output.logits[:, :-1, :], labels[:, 1:])
        seq_log_prob = torch.sum(log_probs[:, input_len:])
    return seq_log_prob.cpu().numpy()
```

While it is clear the alignment between logits and labels (i.e., labels are shifted by 1), it is not clear to me why - when calculating the sequence log probability - we slice the `log_probs` tensor using `input_len` instead of `input_len - 1`. In the example in the book, our overall `output_greedy` sequence is 128 tokens long (it include original input_ids` and, in this case, it stops at `max_length = 128`). However our `log_probs` (after the logits and labels alignment) is a 127-long (dimension 1) tensor. This is because we excluded the last token from the logits and the first token from the labels. As a consequence, we should take this into account when calculating `sequence_logprob` which becomes:

```
def sequence_logprob(model, labels, input_len=0):
    with torch.no_grad():
        output = model(labels)
        log_probs = log_probs_from_logits(
            output.logits[:, :-1, :], labels[:, 1:])
# CHANGE HERE
        seq_log_prob = torch.sum(log_probs[:, input_len-1:])
    return seq_log_prob.cpu().numpy()
```

Am I missing something?
